core/domain/opportunity_domain.py

```diff
@@ -198,3 +198,97 @@ def validate(self):
                 'incomplete_translation, needs_voiceover and assigned_voiceover'
                 ' to be the same as the supported audio languages, '
                 'received %s' % list(expected_set_of_all_languages))
+
+
+class SkillOpportunity(python_utils.OBJECT):
+    """The domain object for skill opportunities."""
+
+    def __init__(
+            self, skill_id, skill_description,
+            question_count=0, topic_id=None, topic_name=None):
+        """Constructs a SkillOpportunity domain object.
+
+        Args:
+            skill_id: str. The unique id of the skill.
+            skill_description: str. The title of the skill.
+            question_count: int. The total number of questions for the skill.
+            topic_id: str. The unique id of the topic.
+            topic_name: str. The name of the topic.
+        """
+        self.topic_id = topic_id
```

Comment:
What happens in the case a skill is assigned to multiple topics?

(Just to confirm -- @aks681 -- that case can occur, right? E.g. "Converting a fraction to decimal" may live under both "Fractions" and "Decimals".)

Reply:
For that case, the current logic is to create a SkillOpportunity for each skill/topic pair.

Reply:
I think this is addressed by the top-level comments on this PR, but please let me know if you have any further questions!